katsdpcontroller/product_controller.py

```diff
@@ -1329,6 +1380,26 @@ def must_wait(node):
             ready.set()
             await shutdown_task
 
+    async def set_delays_impl(
+            self,
+            stream: product_config.GpucbfAntennaChannelisedVoltageStream,
+            timestamp: aiokatcp.Timestamp,
+            coefficient_sets: Sequence[str]) -> None:
+        n_inputs = len(stream.src_streams)
+        conns = []
+        for i in range(n_inputs // 2):
+            node = self._nodes[f"f.{stream.name}.{i}"]
+            if node.katcp_connection is None:
+                raise FailReply(f"No katcp connection to {node.name}")
+            conns.append(node.katcp_connection)
```

Comment:
So the one reservation that I'd have with this, is that if a single f-host wasn't connected, then the entire `?delays` request fails. Is that behaviour that we want?

Reply:
I think it's reasonable until we have health sensors. If we're unable to communicate a delay model to an engine, it should be marked as unhealthy (and maybe eventually killed and restarted).